### PR TITLE
Check if poh recorder has over stepped the leader slot

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
The Poh recorder can potentially tell replay stage to be leader when it has overshoot it's leader slot

#### Summary of Changes
Check for slot boundary to make sure the current slot is still a valid leader slot

### CODE REVIEW

File: core/src/banking_stage.rs [hunk 1]
```diff
@@ -459,6 +460,7 @@ pub fn create_test_recorder(
         bank.slot(),
         Some(4),
         bank.ticks_per_slot(),
+        Pubkey::default(),
```
Comment (rob-solana) [inline core/src/banking_stage.rs]: please pass Pubkeys by ref

Comment (pgarg66) [reply core/src/banking_stage.rs]: 👍

File: core/src/poh_recorder.rs [hunk 2]
```diff
@@ -47,11 +49,21 @@ pub struct PohRecorder {
...
     start_leader_at_tick: Option<u64>,
     max_last_leader_grace_ticks: u64,
+    id: Pubkey,
 }
 
 impl PohRecorder {
     pub fn clear_bank(&mut self) {
-        self.working_bank = None;
+        if let Some(bank) = &self.working_bank {
+            let next_leader_slot =
```
Comment (rob-solana) [inline core/src/poh_recorder.rs]: bank.bank...

File: core/src/poh_recorder.rs [hunk 3]
```diff
@@ -47,11 +49,21 @@ pub struct PohRecorder {
     sender: Sender<WorkingBankEntries>,
     start_leader_at_tick: Option<u64>,
     max_last_leader_grace_ticks: u64,
+    id: Pubkey,
 }
 
 impl PohRecorder {
     pub fn clear_bank(&mut self) {
-        self.working_bank = None;
+        if let Some(bank) = &self.working_bank {
```
Comment (rob-solana) [inline core/src/poh_recorder.rs]: self.working_bank.replace(None) ?

**Pull request merged**